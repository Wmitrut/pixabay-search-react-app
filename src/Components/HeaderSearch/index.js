const HeaderSearch = (props) => {
  return (
    <header className="flex flex-row justify-center items-center p-2 border-b-2 border-indigo-100 2">
      <form onSubmit={props.handleOnSubmit}>
        <input
          ref={props.inputRef}
          focus="true"
          className="pr-10 pl-4 w-full h-10 text-sm bg-white rounded-lg border-2 border-indigo-500/50 hover:border-indigo-500/80 focus:border-indigo-500  focus:ring-0 shadow-sm sm:w-80"
          id="search"
          placeholder="Type away any thing you want to search"
          defaultValue={props.searchTerm}
          aria-label="Search"
        />
        <div className={'mt-2'}>
          <button
            type="button"
            onClick={props.handleOnSubmit}
            className="py-2 px-4 ml-2 font-bold text-white bg-indigo-500 hover:bg-indigo-700 disabled:bg-indigo-500/20 rounded">
            Search
          </button>{' '}
          <button
            type="button"
            onClick={props.handleOnReset}
            disabled={props.searchTerm == ''}
            className="py-2 px-4 ml-2 font-bold text-white bg-indigo-400 hover:bg-indigo-500 disabled:bg-indigo-500/20 rounded">
            Clear Search
          </button>
        </div>
      </form>
    </header>
  );
};

export default HeaderSearch;
