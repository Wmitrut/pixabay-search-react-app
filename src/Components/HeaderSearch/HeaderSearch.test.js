import renderer from 'react-test-renderer';

import HeaderSearch from './index';

it('Renders correctly without data', () => {
  const tree = renderer.create(<HeaderSearch />).toJSON();
  expect(tree).toMatchSnapshot();
});
