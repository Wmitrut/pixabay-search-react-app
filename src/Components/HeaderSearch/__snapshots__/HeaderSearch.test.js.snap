// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly without data 1`] = `
<header
  className="flex flex-row items-center justify-center border-b-2 border-indigo-100 p-2"
>
  <form>
    <input
      aria-label="Search"
      className="h-10 w-full rounded-lg border-2 border-indigo-500/50 bg-white pr-10 pl-4 text-sm shadow-sm hover:border-indigo-500/80  focus:border-indigo-500 focus:ring-0 sm:w-80"
      focus="true"
      id="search"
      placeholder="Type away any thing you want to search"
    />
    <div
      className="mt-2 flex items-center justify-center"
    >
      <button
        className="ml-2 rounded bg-indigo-500 py-2 px-4 font-bold text-white hover:bg-indigo-700 disabled:bg-indigo-500/20"
        type="button"
      >
        Search
      </button>
       
      <button
        className="ml-2 rounded bg-indigo-400 py-2 px-4 font-bold text-white hover:bg-indigo-500 disabled:bg-indigo-500/20"
        disabled={false}
        type="button"
      >
        Clear Search
      </button>
    </div>
  </form>
</header>
`;
